test/e2e/framework/pod/resource.go

```diff
@@ -298,6 +298,26 @@ func podsRunning(c clientset.Interface, pods *v1.PodList) []error {
 	return e
 }
 
+func podContainerFailed(c clientset.Interface, namespace, podName string, containerIndex int, reason string) wait.ConditionFunc {
```

Comment:
What is this containerIndex user for?

Reply:
I was considering the possibility that a test could spawn a pod with multiple containers, so this would basically specify which one to check for. Niche usecase though.